lib/email/receiver.rb

```diff
@@ -196,7 +196,13 @@ def process_internal
     end
 
     def hidden_reason_id
-      @hidden_reason_id ||= is_spam? ? Post.hidden_reasons[:email_spam_header_found] : nil
+      @hidden_reason_id ||= if is_spam?
+                              Post.hidden_reasons[:email_spam_header_found]
+                            elsif auth_res_action == :hide
+                              Post.hidden_reasons[:email_authentication_result_header]
+                            else
+                              nil
+                            end
```

Comment:
I had initially thought that this would *also* flag the post, but through testing realised this wasn't the case. In fact, the hidden reason isn't actually surfaced anywhere in the UI.

I started going down the rabbit hole of flagging this post as well, or enqueuing it instead, but relatively quickly realised it was going to take some time. I think some method of ensuring that these posts appear in a moderator's reviewables is important, but a bit beyond the scope of this PR - as I feel the same thing should happen to incoming emails flagged as spam.

What should we do here?

Reply:
I agree. It might be a good idea to create reviewables for hidden posts in an additional PR. @eviltrout might have some hints for you on that topic.

> the hidden reason isn't actually surfaced anywhere in the UI

I noticed that as well, because I wanted to point out that there's a missing translation for the new hidden reason. But yeah, they are not surfaced on the UI, so no translation is missing. I wonder if that will change when reviewables are created.